### DISCUSSION TITLE
Data Change when server is experiencing issues - surprising result

### DISCUSSION THREAD

**Discussion by david-orn** [FreeOpcUa/opcua-asyncio#1604] [Q&A] [answered]
I am wondering if the following is a bug or a feature:

My objective is to asynchronously monitor for data changes on the server. I am subscribing to a list of nodes using the following method. where `Node` is a valid `asyncua.Node` object:
```
subscription = await node.create_subscription(250, handler)
```

What happened is that the server in question was having some issues. I confirmed this by trying to read directly from the node using
```
value = await node.read_value() 
```

Which resulted in an error:

```
asyncua.ua.uaerrors._auto.BadNoCommunication: Communication with the data source is defined, but not established, and there is no last known value available.(BadNoCommunication)
```

The issue I had was that the subscription method didn't throw a warning or inform that there was an issue with the node. What happened is that within the `datachange_notification` the node was empty `value=0, VariantType=0`. 

It took me a while to understand this behavior. There are two issues
1. Why isn't the `BadNoCommunication` error raised as a warning (on client side) when subscribing
2. Why does the `datachange_notification` get triggered on `create_subscription` even if there is no change in the data

Many thanks

**Comment by oroulet** [comment, marked as answer]
Are you sure the error code was not set on the data value you received in subscription?

**Reply by david-orn** [reply]
Good question. I didn't validate the error code in the data value. 
It's not immediately obvious that additional validation is required on the data value.  I'll add that to help capture the issue.
